lib/Sema/TypeCheckDeclPrimary.cpp

```diff
@@ -2147,6 +2147,27 @@ class DeclChecker : public DeclVisitor<DeclChecker> {
           } else {
             VD->diagnose(diag::property_does_not_override, isClassContext)
                 .highlight(OA->getLocation());
+            //If the override is a property-decl, check if there exists
+            //a no-args func-decl in the base class, if yes, provide a fix-it.
+            auto declName = VD->getBaseName();
+            auto declType = VD->getResultTypeRepr();
+            auto inherited = DC->getSelfClassDecl()->getInherited();
+            //If there is an inherited context, perform member lookup.
+            if (!inherited.empty()) {
+              auto superClass = inherited.front().getType();
+              auto lookupResult = superClass->getAnyNominal()->lookupDirect(declName);
+              
+              for (auto& candidate : lookupResult) {
+                //If the parameter list is empty, compare the return types
+                //of the candidates with the declaration type.
+                if (getParameterList(candidate)->size() == 0) {
+                  auto candidateReturnType = candidate->getResultTypeRepr();
+                  if (candidateReturnType->getKind() == declType->getKind())
+                    VD->diagnose(diag::override_property_in_lieu_of_method, declName.getIdentifier())
+                        .fixItReplace(VD->Decl::getSourceRangeIncludingAttrs(), "override func " + declName.getIdentifier().str().str() + "() -> " + VD->getInterfaceType()->getRValueType().getString() + " { <#code#> }");
```

Comment:
You cannot replace the whole declaration, what you need to do is to find location of `var` and replace that then get location of where the type starts (after `:`) and replace that with your new function type.